src/Spotlight.php

```diff
@@ -82,14 +82,16 @@ public function execute(string $commandId, array $dependencies = []): void
     public function render(): View | Factory
     {
         return view('livewire-ui-spotlight::spotlight', [
-            'commands' => collect(self::$commands)->map(function (SpotlightCommand $command) {
-                return [
-                    'id' => $command->getId(),
-                    'name' => $command->getName(),
-                    'description' => $command->getDescription(),
-                    'dependencies' => $command->dependencies()?->toArray() ?? [],
-                ];
-            }),
+            'commands' => collect(self::$commands)
+                ->filter(fn (SpotlightCommand $command) => $command->shouldBeShown())
```

Comment:
Maybe we can add support for dependency injection.

```suggestion
                ->filter(fn (SpotlightCommand $command) => ImplicitlyBoundMethod::call(app(), [$command, 'shouldBeShown']);)
```

So someone could do something like:

```php
public function shouldBeShown(Request $request): bool
{
    return $request->user()->check();
}
```

What do you think?

Reply:
@PhiloNL Ah good call. Didn't even think of that. I've pushed some changes for that. Let me know if there's anything else you'd like to see on this.

Reply:
Thanks @dvanscott 🙌  one minor change is required to merge this